extended/shared.ts

```diff
@@ -1,9 +1,140 @@
-import type { BNode } from '../b+tree';
-import BTree from '../b+tree';
+import BTree, { BNode } from '../b+tree';
 
-export type BTreeWithInternals<K, V> = {
+/**
+ * BTree with access to internal properties.
+ * @internal
+ */
+export type BTreeWithInternals<K, V, TBTree extends BTree<K, V> = BTree<K, V>> = {
   _root: BNode<K, V>;
   _size: number;
   _maxNodeSize: number;
   _compare: (a: K, b: K) => number;
-} & Omit<BTree<K, V>, '_root' | '_size' | '_maxNodeSize' | '_compare'>;
+} & Omit<TBTree, '_root' | '_size' | '_maxNodeSize' | '_compare'>;
+
+/**
+ * Alternating list storing entries as `[A0, B0, A1, B1, ...]`.
+ * @internal
+ */
+export type AlternatingList<A, B> = Array<A | B>;
+
+/**
+ * Builds leaves from the given alternating list of entries.
+ * The supplied load factor will be respected if possible, but may be exceeded
+ * to ensure the 50% full rule is maintained.
+ * Note: if < maxNodeSize entries are provided, only one leaf will be created, which may be underfilled.
+ * @param alternatingList The list of entries to build leaves from.
+ * @param maxNodeSize The maximum node size (branching factor) for the resulting leaves.
+ * @param onLeafCreation Called when a new leaf is created.
+ * @param loadFactor Desired load factor for created leaves. Must be between 0.5 and 1.0.
+ * @internal
+ */
+export function makeLeavesFrom<K, V>(
+  alternatingList: AlternatingList<K, V>,
```

Comment:
I wonder if this is the best interface, because if instead you accepted two arrays K[] and V[], you could extract subarrays with `slice()` which (having a C/C++ impl) is likely a faster way to construct the `BTree`.